### DISCUSSION TITLE
What's the difference between this repo and rules_apple?

### DISCUSSION THREAD

**Discussion by bartekpacia** [MobileNativeFoundation/rules_xcodeproj#2836] [Q&A] [answered]
Hi,

Thank you for the amazing, open-source work on this repo. It all looks very cool.

I'm new to Bazel and researching its ecosystem. So far I haven't been able to find a clear explanation between use cases of `rules_apple` and `rules_xcodeproj`? It seems to me like they're both similar, but `rules_xcodeproj` depends on `rules_apple`. I also see that `rules_apple` had existed long before `rules_xcodeproj` was ever created.

I'll be grateful for an explanation, and thank you again.

**Comment by brentleyjones** [comment, marked as answer]
**rules_apple** provides you with rules to create targets for Apple platforms, e.g. iOS Application bundles. **rules_xcodeproj** provides you with a rule that creates an Xcode project file (`.xcodeproj`) that builds your Bazel targets with Bazel. You can use **rules_xcodeproj** without using any **rules_apple** rules, but most likely you will have and use some if you are using **rules_xcodeproj**.
